File: Python/Dosen.py
# Import Class yang digunakan
from SivitasAkademik import SivitasAkademik
from ProgramStudi import ProgramStudi

class Dosen(SivitasAkademik):

    # Kelas Dosen di sini berperan sebagai:
    # - Child dari Kelas SivitasAkademik
    # - GrandChild dari Kelas Human

    # ATRIBUT
    __NIP = ""
    __pend_terakhir = ""
    __keahlian = ""

    # METHODS
    # Constructor
    def __init__(self, nik="", nama="", jk='',  asal_univ="", email_edu = "", fakultas="", prodi = ProgramStudi(), nip = "", pend_terakhir = "", keahlian = ""):
        super(Dosen, self).__init__(nik, nama, jk, asal_univ, email_edu, fakultas, prodi)
        self.__NIP = nip
        self.__pend_terakhir = pend_terakhir
        self.__keahlian = keahlian

    # Setter and Getter
    def getNIP(self):
        return self.__NIP
    
    def setNIP(self, nip):
        self.__NIP = nip

    def getPendidikanTerakhir(self):
        return self.__pend_terakhir
    
    def setPendidikanTerakhir(self, pend=""):
        self.__pend_terakhir = pend

    def getKeahlian(self):
        return self.__keahlian
    
    def setKeahlian(self, keahlian):
        self.__keahlian = keahlian

File: CPP/Program/dosen.cpp
#include "header.hh"

// == Kelas Dosen ==

// Constructors
Dosen::Dosen()
{
    this->NIP = "";
    this->pend_terakhir = "";
    this->keahlian = "";
}

Dosen::Dosen(string NIP, string pend_terakhir, string keahlian)
{
    this->NIP = NIP;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
}

Dosen::Dosen(string nik, string nama, char jk, string asal_universitas, string email_edu, string fakultas, ProgramStudi *prodi, string nip, string pend_terakhir, string keahlian) : SivitasAkademik(nik, nama, jk, asal_universitas, email_edu, fakultas, prodi)
{
    this->NIP = nip;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
}

// Getter and Setter
string Dosen::getNIP()
{
    return this->NIP;
}
string Dosen::getPendidikanTerakhir()
{
    return this->pend_terakhir;
}
string Dosen::getKeahlian()
{
    return this->keahlian;
}

void Dosen::setNIP(string nip)
{
    this->NIP = nip;
}
void Dosen::setPendidikanTerakhir(string pend_terakhir)
{
    this->pend_terakhir = pend_terakhir;
}
void Dosen::setKeahlian(string keahlian)
{
    this->keahlian = keahlian;
}

// Destructors
Dosen::~Dosen()
{
}